sdk/python/tests/compiler/compiler_tests.py

```diff
@@ -423,6 +423,38 @@ def my_pipeline(a: {'GCSPath': {'path_type':'file', 'file_type': 'csv'}}='good',
     finally:
       shutil.rmtree(tmpdir)
 
+  def test_type_checking_with_json_schema(self):
+    """Test type check pipeline parameters against the json schema."""
+    @component
+    def a_op(field_m: {'GCRPath': {'openapi_schema_validator': {"type": "string", "pattern": "^.*gcr\\.io/.*$"}}}, field_o: 'Integer'):
+      return ContainerOp(
+          name = 'operator a',
+          image = 'gcr.io/ml-pipeline/component-b',
+          arguments = [
+              '--field-l', field_m,
+              '--field-o', field_o,
+          ],
+      )
+
+    @pipeline(
+        name='p1',
+        description='description1'
```

Comment:
JFYI, name and description are optional now.

Reply:
Thanks.